src/flow.ts

```diff
@@ -806,6 +837,59 @@ export class Flow {
         thisLocalFlags[i] = newFlags;
       }
     }
+
+    this.mergeFieldFlags(left, right);
+  }
+
+  /**
+   * Merges the fields flags of the given flows
+   * into the current flow. Flags will only be merged
+   * if both flows definitely define the flags.
+   */
+  mergeFieldFlags(left: Flow, right: Flow): void {
+    if (left.fieldFlags !== null &&
+      right.fieldFlags !== null &&
+      right.fieldFlags.size > 0
+    ) {
```

Comment:
```suggestion
  mergeFieldFlags(left: Flow, right: Flow): void {
    if (
      left.fieldFlags !== null &&
      right.fieldFlags !== null &&
      right.fieldFlags.size > 0
    ) {
```

Reply:
Also plz remove `const leftFieldFlags = left.fieldFlags;` and `const rightFieldFlags = right.fieldFlags;` inside if block later

Reply:
I mean [here](https://github.com/AssemblyScript/assemblyscript/pull/1155/files#diff-8d2c019b40813ce77885d2ad54c2cb15R857) and [here](https://github.com/AssemblyScript/assemblyscript/pull/1155/files#diff-8d2c019b40813ce77885d2ad54c2cb15R861)

Reply:
Right, good catch.